packages/data-context/src/actions/WizardActions.ts

```diff
@@ -84,75 +90,33 @@ export class WizardActions {
 
   async initialize () {
     if (this.ctx.currentProject) {
-      this.data.detectedFramework = null
-      this.data.detectedBundler = null
-      this.data.detectedLanguage = null
+      this.ctx.update((coreData) => {
+        coreData.wizard.detectedFramework = null
+        coreData.wizard.detectedBundler = null
+        coreData.wizard.detectedLanguage = null
+      })
 
       await this.detectLanguage()
       debug('detectedLanguage %s', this.data.detectedLanguage)
       this.data.chosenLanguage = this.data.detectedLanguage || 'js'
 
-      let hasPackageJson = true
+      const detected = detect(await fs.readJson(path.join(this.ctx.currentProject, 'package.json')))
 
-      try {
-        await this.ctx.fs.access(path.join(this.ctx.currentProject, 'package.json'), this.ctx.fs.constants.R_OK)
-      } catch (e) {
-        debug('Could not read or find package.json: %O', e)
-        hasPackageJson = false
-      }
-      const packageJson: {
-        dependencies?: { [key: string]: string }
-        devDependencies?: { [key: string]: string }
-      } = hasPackageJson ? await this.ctx.fs.readJson(path.join(this.ctx.currentProject, 'package.json')) : {}
-
-      debug('packageJson %O', packageJson)
-      const dependencies = [
-        ...Object.keys(packageJson.dependencies || {}),
-        ...Object.keys(packageJson.devDependencies || {}),
-      ]
-
-      this.detectFramework(dependencies)
-      debug('detectedFramework %s', this.data.detectedFramework)
-      this.detectBundler(dependencies)
-      debug('detectedBundler %s', this.data.detectedBundler)
-
-      this.data.chosenFramework = this.data.detectedFramework || null
-      this.data.chosenBundler = this.data.detectedBundler || null
-    }
-  }
-
-  private detectFramework (dependencies: string[]) {
```

Comment:
All this detection stuff moved to `@packages/scaffold-config` and is covered head-to-toe with tests.